src/modules/patient/providers/patient-query.service.ts

```diff
@@ -67,4 +74,99 @@ export class PatientQueryService extends TypeOrmQueryService<Patient> {
         return patients;
     }
 
+
+    async getQuestionnaireReport(id: number, status: string, questionnaireId: string): Promise<PatientReport> {
+        const patient = await this.repo.findOne({
+            join: { alias: 'patient', innerJoinAndSelect: { assessments: 'patient.assessments' } },
+
+            where: qb => {
+                qb.where({
+                    id
+                }).andWhere('assessments.status = :status', { status });
+            },
+        }
+        );
+        const queries = [] as Promise<QuestionnaireAssessment>[];
+        for (const assessment of patient.assessments) {
+            queries.push(this.questionnaireAssessmentService.getById(assessment.questionnaireAssessmentId))
+        }
+        const questionnaireAssessment = await Promise.all(queries);
+        let answeredQuestionnaire;
```

Comment:
In-case patient can have answered same questionnaire more than once I would recommend pluralizing `answeredQuestionnaire` here to `answeredQuestionnaires` and initializing it as an empty array of type `AnsweredQuestionnare[]`

In next part in the for-loop for matched values, they can be pushed to the array and returned as part of the result

Reply:
The answered questionnaire will be filtered from front-end, so it will always be only 1

Reply:
I think it could be possible a single questionnaire could be responded multiple times, thus same questionaireId could have multiple assessment responses for same patient. I think we can crosscheck this with @edisoni1995 to be sure